holopy/core/utils.py

```diff
@@ -39,6 +39,29 @@
 
 from holopy.core.errors import DependencyMissing
 
+
+# FIXME this is difficult to test, since it works on the file level
```

Comment:
@barkls  and @ralex0 -- I don't know how to test this effectively. There are ways of testing a normal print statement (e.g. contextlib.redirect_stdout), but I can't find one that works for output generated by external code. Do either of you have any thoughts?

Reply:
Is the problem that the fortran code is outputting too much information?  If so, we should check to see if the code is outputting to stdout or stderr.  By UNIX convention, diagnostic information should be printed to stderr rather than stdout.  Then you can filter the correct stream (or change the fortran code to output to stderr).

Reply:
The problem is just that the fortran code outputs extraneous info, which makes it more difficult to see which tests have passed and failed (among other things). It's just a fortran ```print*``` statement (line 146ff of ```holopy.scattering.theory.mie_f.scsmfo_min.for```) so I assume it's outputting to stdout. I don't think stderr is the place for it, since it's not really error related, it's just ... noise.

I'd be happy to edit the fortran code and delete the offending lines, but I wasn't sure if other holopy users use them, so I figured I'd just use that context manager to pipe the output to devnull.

Do you think it's better to delete the print statements from the fortran code, or leave them there and capture it?

Reply:
I agree stderr isn't appropriate. I have never made use of the terminal output from the multisphere code when I run it. In fact, I had those lines commented out of my local fortran file for a long time until I wrote the current code that suppresses it, so I'd definitely be on board to delete them.

Reply:
stderr is the appropriate stream.  stdout is not.  Any output from the program that doesn't go directly into the stdin for another program should be directed to stderr.  That includes diagnostic information.  We should not delete the lines from the fortran code that output this information.  It's not noise -- it gives information about the convergence of the conjugate gradient solver.  But you can direct it to stderr and then redirect to /dev/null if you don't want it.